client/tx/search.go

```diff
@@ -105,7 +105,8 @@ $ gaiacli query txs --tags '<tag1>:<value1>&<tag2>:<value2>' --page 1 --limit 30
 // SearchTxs performs a search for transactions for a given set of tags via
 // Tendermint RPC. It returns a slice of Info object containing txs and metadata.
 // An error is returned if the query fails.
-func SearchTxs(cliCtx context.CLIContext, cdc *codec.Codec, tags []string, page, limit int) ([]sdk.TxResponse, error) {
+func SearchTxs(cliCtx context.CLIContext, cdc *codec.Codec,
+	tags []string, page, limit int) (*sdk.SearchTxsResult, error) {
```

Comment:
Why are we returning a pointer to `SearchTxsResult`?

Reply:
For compatibility with `return nil, error.New("...")`
```go
        if len(tags) == 0 {
		return nil, errors.New("must declare at least one tag to search")
	}		
```